src/lib/repository.js

```diff
@@ -63,7 +63,8 @@ module.exports = async function(pkg, info) {
       },
     ]);
     info.ghepurl = answers.url;
-    return;
+    if (answers.enterprise) return;
+    throw new Error('Invalid GitHub repository URL');
```

Comment:
I think this error might be confusing? If I answer the the enterprise question with no, then the "Invalid GitHub repository URL" error will get thrown? Is that intended? Or am I missing something?

Reply:
@gr2m How about a different error message such as: "Failed to detect a valid Github repository URL. Please verify your configuration."? Currently, if you answer "no" to the Github Enterprise prompt when your project contains an invalid Git URL, a null value for the parsed repository URL is passed all the way to the CI-specific configuration stage (https://github.com/semantic-release/cli/blob/master/src/index.js#L86) before failing with the TypeError mentioned in #100.

Reply:
sounds like the problem is the statement above:

```js
info.ghepurl = answers.url;
```

Maybe wrapping it in an if block where we check if the enterprise question was answered with yes?

```js
if (answers.enterprise) {
  info.ghepurl = answers.url
}
```

Reply:
The assignment causing the TypeError is `info.ghrepo = {slug: parsedUrl};` (https://github.com/semantic-release/cli/blob/master/src/lib/repository.js#L69) since `parsedUrl` is still null after the GitHub Enterprise prompt if you answer `no`.

Reply:
Ah okay now I understand. A better error message will do then, maybe add the `repoUrl` to the message?

```js
throw new Error(`GitHub repository URL is invalid: ${repoUrl}`)
```

Reply:
Sure